src/adapter/http/src/data/metadata_handler.rs

```diff
@@ -156,6 +157,8 @@ pub async fn dataset_metadata_handler(
         .await
         .api_err()?;
 
+    check_dataset_read_access(&catalog, resolved_dataset.get_handle()).await?;
```

Comment:
Discussion of the approach: 
- Overall, I like the implemented approach, handlers responsible for verifying access 
- Alternative approach: `DatasetAuthorizationMiddleware` connect for handlers as well, since now the middleware is only responsible for accesses

Please share what you think about this (the alternative approach)?

Reply:
Normally it should not be a handler responsibility.

Reply:
`DatasetAuthorizationMiddleware` was used to be responsible for access validation

Done